### ISSUE TITLE
Use URL and FileBlob/WebBlob to load binary data in Inference API instead of Blob

### ISSUE THREAD

**Issue by Aschen** [huggingface/huggingface.js#114]
The Inference API use Blob or ArrayBuffer as argument for the request who result in loading huge file into RAM before sending them to the API.

We should rather use the new FileBlob or LazyBlob as for the `commit` method of the Hub API.

I already started to externalize the first version of the `FileBlob` https://github.com/aschen/lazyblob

I would also understand if you want to:
  1) keep the package in the `huggingface` namespace 
  2) create a shared package between `hub` and `inference` package in this monorepo.

**Comment by coyotte508** [comment]
I think publishing the package is neat and you did most of the work, so definitely go ahead! It'll be useful to the community / nice to show off :).

Still, for our usage we'll want to keep it in an internal local internal package.

**Comment by Aschen** [comment]
This work was teamwork and I had good advices on a topic I rarely encounter ;-)

For our matter, I suggest to create an internal package then (if later I have more time then I will publish)

Shall I create a `packages/shared` for the code shared between both libraries?

**Comment by coyotte508** [comment]
I guess `packages/utils`? Taking inspiration from https://github.com/typescript-eslint/typescript-eslint/tree/main/packages/utils

Let's do this after merging the two PRs (web blobs & VCR), just in case it interferes

**Comment by Aschen** [comment]
Sounds good :+1: 

Do we really need to publish it as a separate package on NPM?

The alternative would be to import the classes directly from a relative path. Additional chunks will be added for the code coming from the `packages/utils` package

![image](https://user-images.githubusercontent.com/4447392/228095043-79339397-ed28-4b97-afc6-ea0b218cb6b5.png)

**Comment by coyotte508** [comment]
Yes I don't want to add it as a separate package, either importing the files directly works, or changing the `tsup` config (or associated `rollup` config)

**Comment by Aschen** [comment]
:100: agree with KISS :+1: 

I wil try something

**Comment by coyotte508** [comment]
Note that you can add internal dependencies in pnpm with `workspace:*`, just they will appear in the release's package.json by default

**Comment by coyotte508** [comment]
And if it's too much headache, a shared folder like you suggested is :+1:

**Comment by Aschen** [comment]
> Note that you can add internal dependencies in pnpm with `workspace:*`, just they will appear in the release's package.json by default

I'm not sure to understand what you mean. 

For instance I tried to create a new package `@huggingface/utils` and then import it in `hub` but then the generated code still had my import, expecting it to be published on NPM as well.

**Comment by ArsalaBangash** [comment]
Hi, I'm working on a tutorial for Scrimba and I did a quick search for arraybuffer in the issues because due to the library calling blob.arrayBuffer, it doesn't work inside webcontainers. 

Is this related or should I open a new issue?

**Comment by coyotte508** [comment]
no this issue is old and can be closed

you can open a new one with details

**Comment by ArsalaBangash** [comment]
@coyotte508 I managed to get the arrayBuffer problem resolved. No need for a new issue 🙌
